### PR TITLE
fix the decision of Filter

### PR DESCRIPTION
**Pull request by siyuhong**
Hi, 
I think the filter logic is false，now I fixed it. 
And, I add case sensitivity able in Stringmatchfilter.
Please review this code if you have time.

### CODE REVIEW

File: src/log4qt/varia/levelmatchfilter.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ LevelMatchFilter::LevelMatchFilter(QObject *parent) :
 Filter::Decision LevelMatchFilter::decide(const LoggingEvent &event) const
 {
     if (mLevelToMatch == Level::NULL_INT ||
-            event.level() != mLevelToMatch)
+            event.level() == mLevelToMatch)
```
Comment (andibacher) [inline src/log4qt/varia/levelmatchfilter.cpp:37]: Why do you think that if the level matches there should be returned NEUTRAL? Wenn the level matches there should be returned ACCEPT!

Comment (andibacher) [reply src/log4qt/varia/levelmatchfilter.cpp:37]: That is the log4j documentation of of NEUTRAL, ACCEPT, DENY:
NEUTRAL: This filter is neutral with respect to the log event. The
remaining filters, if any, should be consulted for a final decision.

ACCEPT : The log event must be logged immediately without consulting with
the remaining filters, if any, in the chain. 

DENY: The log event must be dropped immediately without consulting
 with the remaining filters, if any, in the chain.

File: src/log4qt/varia/stringmatchfilter.h [hunk 2]
```diff
@@ -59,13 +59,14 @@ class LOG4QT_EXPORT StringMatchFilter : public Filter
     bool acceptOnMatch() const;
     QString stringToMatch() const;
     void setAcceptOnMatch(bool accept);
-    void setStringToMatch(const QString &string);
+    void setStringToMatch(const QString &string, Qt::CaseSensitivity cs = Qt::CaseInsensitive);
```
Comment (andibacher) [inline src/log4qt/varia/stringmatchfilter.h:62]: This is an extensions that can be accepted.

**Review by andibacher — CHANGES_REQUESTED** [review]

File: src/log4qt/varia/levelmatchfilter.cpp (hunk 1 above)
Comment (andibacher) [reply src/log4qt/varia/levelmatchfilter.cpp:37]: Unit Test are also failing because of this change.

Comment (siyuhong) [reply src/log4qt/varia/levelmatchfilter.cpp:37]: Sorry，I compared the log4j code and you were right. 
But I have a question: it is still append by default when all levelmatchfilter are non-matched (NEUTRAL)
`
    Filter  *filter = mpHeadFilter.data();
    while (filter)
    {
        Filter::Decision decision = filter->decide(event);
        if (decision == Filter::ACCEPT)
            break;
        else if (decision == Filter::DENY)
            return;
        else
            filter = filter->next().data();
    }
    append(event);
`

**Review by andibacher — CHANGES_REQUESTED** [review]

File: src/log4qt/varia/levelmatchfilter.cpp (hunk 1 above)
Comment (andibacher) [reply src/log4qt/varia/levelmatchfilter.cpp]: I don't know why this is the behavior but it is the same as in log4j 1:

`    Filter f = this.headFilter;
    
    FILTER_LOOP:
    while(f != null) {
      switch(f.decide(event)) {
      case Filter.DENY: return;
      case Filter.ACCEPT: break FILTER_LOOP;
      case Filter.NEUTRAL: f = f.getNext();
      }
    }
    
    this.append(event);  `

Comment (siyuhong) [reply src/log4qt/varia/levelmatchfilter.cpp]: > I don't know why this is the behavior but it is the same as in log4j 1:
> 
> ` Filter f = this.headFilter;
> 
> ```
> FILTER_LOOP:
> while(f != null) {
>   switch(f.decide(event)) {
>   case Filter.DENY: return;
>   case Filter.ACCEPT: break FILTER_LOOP;
>   case Filter.NEUTRAL: f = f.getNext();
>   }
> }
> 
> this.append(event);  `
> ```

OK，Thanks for you reply.

File: src/log4qt/varia/stringmatchfilter.h (hunk 2 above)
Comment (andibacher) [reply src/log4qt/varia/stringmatchfilter.h:62]: Please extend the test cases to verify CaseSensitivity and CaseInsensitivty of the StringMatchFilter data driven test of the log4qttest. Thanks,

Comment (siyuhong) [reply src/log4qt/varia/stringmatchfilter.h]: > 

OK, I have committed code.

**Pull request merged**